proxy/src/proxy.rs

```diff
@@ -206,11 +207,12 @@ impl Proxy<ConnectedState> {
                 }
             }
         };
-        trace!(greeting=%format!("{:?}", greeting).blue(), role = "s2p", "<--| Received greeting");
+        let msg = format!("{:?}", greeting).blue();
```

Comment:
This forces strict evaluation (i.e. the string is created even if the logger is disabled). Do we want to accept that? It might cause large overhead if the proxy is used in production.

Same for all other cases.

Reply:
Good catch! No. It was an oversight and I want to fix this before merging.